### ISSUE TITLE
Adding `TabViewer::closable(..)` ? 

### ISSUE THREAD

**Issue by kang-sw** [Adanos020/egui_dock#113]
Right now, it doesn't seem possible to set the close button for all widgets in bulk, and dynamically set whether or not to render the close button for individual widgets...

When configuring a dock, some widgets may not define a closing behaviour. Not handling the `TabViewer::on_close()` call serves this purpose, but it would be nice to add a `TabViewer::closable()` function so that we can avoid defining a closing behaviour for objects that can't be closed, so that the close button itself doesn't appear in the tab bar and context menu.

**from**

```rust
fn __(vwr: &mut TabViewer, style: &Style, tab: &mut Tab) {
  if style.show_close && close_btn(ui) && vwr.on_close(tab)  {
    /* do something closinbg here */
  }
}
```

**to**
```rust
fn __(vwr: &mut TabViewer, style: &Style, tab: &mut Tab) {
  if style.show_close && vwr.closable(tab) && close_btn(ui) && vwr.on_close(tab) {
    /* do something closinbg here */
  }
}
```

**Comment by Adanos020** [comment]
Sounds good to me, seems to be a simple and non-breaking change. Should be fine to get this directly in the main branch.

**Comment by ptxmac** [comment]
I took a quick stab at this in #150

**Comment by Adanos020** [comment]
Closed by #150
